sources/packages/backend/apps/queue-consumers/src/processors/schedulers/sfas-integration/_tests_/sfas-integration.scheduler.e2e-spec.ts

```diff
@@ -0,0 +1,177 @@
+    sftpClientMock = sshClientMock;
+    // Processor under test.
+    processor = app.get(SFASIntegrationScheduler);
+    systemUsersService = nestApplication.get(SystemUsersService);
+    // Create student if it doesn't exist.
+    studentExists = await db.student.findOne({
+      where: {
+        birthDate: getISODateOnlyString(new Date("1998-03-24")),
+        user: { lastName: "FOUR" },
+        sinValidation: { sin: "900041310" },
+      },
+    });
+    sharedStudent = studentExists;
+    if (!studentExists) {
+      sharedStudent = await saveFakeStudent(db.dataSource);
+      // Update the student to ensure that the student imported from SFAS is the same student as the one created above.
+      sharedStudent.birthDate = getISODateOnlyString(new Date("1998-03-24"));
+      sharedStudent.user.lastName = "FOUR";
+      sharedStudent.sinValidation.sin = "900041310";
+      await db.student.save(sharedStudent);
+    }
+  });
+
+  beforeEach(async () => {
+    jest.clearAllMocks();
+    // Set the isActive field to false for each record of the sharedStudent in the student restrictions table to not interfere with other tests.
+    await db.studentRestriction.update(
+      { student: { id: sharedStudent.id } },
+      { isActive: false },
+    );
+  });
+
+  it("Should add only one legacy restriction for the student when one or more restrictions are imported from SFAS which do not have a mapping to the SIMS restrictions", async () => {
+    // Arrange
+    // Queued job.
+    const job = createMock<Job<void>>();
+    mockDownloadFiles(sftpClientMock, [SFAS_LEGACY_RESTRICTION_FILENAME]);
+    // Act
+    await processor.processSFASIntegrationFiles(job);
+    // Assert
+    // Expect the file was deleted from SFTP.
+    expect(sftpClientMock.delete).toHaveBeenCalled();
+    const studentRestrictionsCount = await db.studentRestriction.count({
+      where: {
+        student: { id: sharedStudent.id },
+        restriction: { restrictionCode: RestrictionCode.LGCY },
+        isActive: true,
+      },
+    });
+    const studentRestriction = await db.studentRestriction.findOne({
+      select: { creator: { id: true } },
+      relations: { creator: true },
+      where: {
+        student: { id: sharedStudent.id },
+        restriction: { restrictionCode: RestrictionCode.LGCY },
+        isActive: true,
+      },
+    });
+    // Expect only one Legacy restriction to be inserted.
+    expect(studentRestrictionsCount).toBe(1);
+    // Expect the restriction to be created by a system user.
+    expect(studentRestriction.creator.id).toEqual(
+      systemUsersService.systemUser.id,
+    );
+  });
+
+  it(
+    "Should not insert a restriction to SIMS student restrictions when the same restriction is imported from SFAS but is already active in SIMS or has a restriction resolution date, " +
+      "Should insert a restriction after mapping it to SIMS restriction when it is either inactive or is not at all present in the SIMS student restrictions, ",
```

Comment:
I think we should have distinct test cases separate. The reason I think it should be separate is because if it fails it should be clear which case is failing.

Reply:
I had deliberately kept these scenarios to test them as a part of a single insert through the SFAS import file rather than through multiple files each having their own individual small test scenarios.